### PR TITLE
FIX: uses new addPostAdminMenuButton component API

### PR DESCRIPTION
**Pull request by jjaffeux**
This change allows to properly support this complex case. The button has to make an ajax call to decide if it's valid or not, which was a very bad pattern of the existing code and not possible through the existing API.

Note this commit is also adding support for new linting rules as it made easier to modify the code.

### CODE REVIEW

File: assets/javascripts/discourse/components/post-admin-menu-approve-button.gjs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import Component from "@glimmer/component";
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/post-admin-menu-approve-button.gjs:1]: This is the component used by the new API

File: assets/javascripts/discourse/initializers/category-experts-post-decorator.js [hunk 2]
```diff
@@ -1,34 +1,7 @@
-import { ajax } from "discourse/lib/ajax";
-import { iconNode } from "discourse-common/lib/icon-library";
 import { next } from "@ember/runloop";
-import { popupAjaxError } from "discourse/lib/ajax-error";
 import { withPluginApi } from "discourse/lib/plugin-api";
-
-function setPostCategoryExpertAttributes(
```
Comment (jjaffeux) [inline assets/javascripts/discourse/initializers/category-experts-post-decorator.js:7]: Move in it's own lib

File: assets/javascripts/discourse/initializers/category-experts-post-admin-menu.js [hunk 3]
```diff
@@ -1,60 +1,16 @@
...
-        return helper.attach(
-          "category-experts-post-admin-menu-btn",
-          helper.attrs
-        );
+    withPluginApi("1.16.0", (api) => {
+      api.addPostAdminMenuButton((attrs) => {
+        return {
+          component: PostAdminMenuApproveButton,
+          data: attrs,
+        };
```
Comment (jjaffeux) [inline assets/javascripts/discourse/initializers/category-experts-post-admin-menu.js:13]: The bulk of the change allowed by https://github.com/discourse/discourse/pull/24112

**Review by davidtaylorhq — APPROVED** [review]

**Review by davidtaylorhq — APPROVED** [review]

**Comment by davidtaylorhq** [comment]
Superseded by https://github.com/discourse/discourse-category-experts/pull/126

**Pull request closed without merging**